src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -643,6 +629,7 @@ private static void SubstituteConstraintTypes(ImmutableArray<TypeSymbolWithAnnot
                 if (considerCustomModifiers)
                 {
                     if (!type1.TypeSymbol.Equals(type2.TypeSymbol, TypeCompareKind.ConsiderEverything.AddIgnoreDynamic(ignoreDynamic).AddIgnoreTupleNames(ignoreTupleNames)) ||
+                        !type1.CustomModifiers.SequenceEqual(type2.CustomModifiers) ||
```

Comment:
I believe this should be folded into comparison of type1 against type2,, the previous line should do that instead of comparing only type symbols.